### PR TITLE
[luci/profile] Add CircleNodeOrigin to README.md

### PR DESCRIPTION
**Pull request by llFreetimell**
This commit will add explanation about `CircleNodeOrigin` to `README.md` in `luci/profile`.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

**Comment by llFreetimell** [comment]
This PR is originated from https://github.com/Samsung/ONE/pull/6238#discussion_r590060293, https://github.com/Samsung/ONE/pull/6240#issuecomment-793473486.

After this PR is merged, duplicated comments will be removed in source code.

File: compiler/luci/profile/README.md [hunk 1]
```diff
@@ -1,3 +1,121 @@
...
+   - `node5` is **replaced with newly created** `node6`.
+   - origin of `node5` is copied to origin of `node6`.
+
+**Therefore, when using `CircleNodeOrigin`, please aware of the most important principle. "Preserve origin information"**
+
+Next items are about implementation details to store the origin information.
+
+### Source Table
+
+Source table includes a set of id of origin node and name of origin node.
```
Comment (seanshpark) [inline compiler/luci/profile/README.md:49]: how about
```
... a set of id and name of origin node.
```

File: compiler/luci/profile/README.md [hunk 2]
```diff
@@ -1,3 +1,121 @@
...
+```
+[ entry_number : uint32_t ]
+[ id : uint32_t ][ length : uint32_t ][ data : char * length ] * entry_number
+```
+- entry_number : The number of entry
+  - Each entry consist of id, length, and data.
+- id : ID of origin node
+- length : Length of data
+- data : Name of origin node
+  - **`data` always ends with `\0` character.**
```
Comment (seanshpark) [inline compiler/luci/profile/README.md:62]: how about
```
data : Name of origin node (null-terminated string)
```
as I commented before, please refer https://en.wikipedia.org/wiki/Null-terminated_string

File: compiler/luci/profile/README.md [hunk 3]
```diff
@@ -1,3 +1,121 @@
...
+
+#### In-memory format
+```cpp
+// size = entry_number
+std::map<uint32_t /* id */, std::string /* name */>
+```
+  - **`\0` chracter is not inserted to `std::string`.**
```
[the remark this answers is not in the document]
Comment (seanshpark) [reply compiler/luci/profile/README.md:69]: What happens when you add the last `\0` ?
FYI, please check how this comes out (there maybe seg fault... but let's ignore this)
```cxx
std::string h = "hello world";
const char *c = h.c_str();
for (int i=0; i<32; ++i) printf("%d ", *(c+i));
```

File: compiler/luci/profile/README.md [hunk 4]
```diff
@@ -1,3 +1,121 @@
...
+### Op Table
+
+Op table includes a set of id of operation and id(s) of operation's origin nodes.
+
+#### Binary format
+
+Op table is stored in circle file as binary with following format.
+```
+[ entry_number : uint32_t ]
+[ id : uint32_t ][ node_num : uint32_t ][ node_ids : node_num * uint32_t ] * entry_number
```
Comment (seanshpark) [inline compiler/luci/profile/README.md:95]: how about
```
... node_ids : uint32_t * node_num  ...
```

File: compiler/luci/profile/README.md [hunk 5]
```diff
@@ -1,3 +1,121 @@
...
+```
+- entry_number : The number of entry
```
Comment (seanshpark) [inline compiler/luci/profile/README.md:97]: `entry` or `entries` ?

File: compiler/luci/profile/README.md [hunk 6]
```diff
@@ -1,3 +1,121 @@
...
+  - Each entry consist of id, node_num, and node_ids.
```
Comment (seanshpark) [inline compiler/luci/profile/README.md:98]: `consist` or `consists` ?

File: compiler/luci/profile/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (llFreetimell) [reply compiler/luci/profile/README.md:69]: > FYI, please check how this comes out (there maybe seg fault... but let's ignore this)
> ```cxx
> std::string h = "hello world";
> const char *c = h.c_str();
> for (int i=0; i<32; ++i) printf("%d ", *(c+i));
> ```

Can you let me know what is this for?
I want to know which point should I focus on :)

[the remark this answers is not in the document]
Comment (seanshpark) [reply compiler/luci/profile/README.md:68]: to check if `std::string` contains `\0` at the end of string :)

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**